Content.Client/_CD/WelcomeMessage/WelcomeWindow.xaml.cs

```diff
@@ -0,0 +1,42 @@
+using Content.Client.Guidebook;
+using Content.Client.UserInterface.Controls;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.ContentPack;
+using Robust.Shared.Utility;
+
+namespace Content.Client._CD.WelcomeMessage;
+
+[GenerateTypedNameReferences]
+public sealed partial class WelcomeWindow : FancyWindow
+{
+    [Dependency] private readonly IResourceManager _resourceManager = default!;
+    [Dependency] private readonly DocumentParsingManager _parsingMan = default!;
+    [Dependency] private readonly ILogManager _logManager = default!;
+
+    private readonly ISawmill _sawmill;
+
+    public WelcomeWindow()
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+        _sawmill = _logManager.GetSawmill("_CD.we.WelcomePopup");
+
+        PopupCloseButton.OnPressed += _ =>
+        {
+            Close();
+        };
+    }
+
+    public void LoadContents(ResPath path)
+    {
+        using var file = _resourceManager.ContentFileReadText(path);
+
+        if (!_parsingMan.TryAddMarkup(Contents, file.ReadToEnd()))
+        {
+            Contents.AddChild(new Label() { Text = "ERROR: Failed to parse document." });
+            _sawmill.Error($"Failed to parse contents of guide document {path}.");
+        }
+    }
+}
```

Comment:
ReadToEndAsync?

Reply:
I was unable to figure out how to get that to work (I have only used c# for ss14). 

My attempt consisted of changing LoadContents to async (returning a Task) and awaiting `file.ReadToEndAsync`. I then added an optional Task field to the UI controller and called LoadContents with this snippet:

```csharp
            _loadTask = _window.LoadContents(new ResPath(ContentsPath))
                .ContinueWith(async res =>
            {
                _currentHash = await res;
            });
```

Finally, I waiting on _loadTask before comparing the hashes.

This implementation seemed to cause the game to hang most of the time upon loading in.

Reply:
Is there an example in ss14 that you can point me to? It looks like most other things that read files use the non-async ReadToEnd.